src/ensureApiEnabled.ts

```diff
@@ -68,8 +74,27 @@ async function enable(projectId: string, apiName: string): Promise<void> {
       )} can't be enabled until the upgrade is complete. To upgrade, visit the following URL:
 
 https://console.firebase.google.com/project/${projectId}/usage/details`);
+    } else if (isPermissionError(err)) {
+      // Recognize permission denied errors on APIs and provide users the
+      // GCP console link to easily enable the API.
+      const permissionsError = apiPermissionDeniedRegex.exec((err as Error).message);
+      if (permissionsError && permissionsError[1]) {
```

Comment:
Wont this fail if permissionsError.length <2

Reply:
Yes, but if it fails that means my regex failed and I don't have a capturing group with the string of the API name, so I can't go forward with constructing this message anyway. It will fall back to throwing the original error which should be the most useful thing we have.